engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/html/helper/DefaultHtmlContentGenerator.java

```diff
@@ -259,6 +259,11 @@ public String rewrite( final ImageContainer image, ContentItem contentItem ) {
 
   public ContentItem writeImage( final ImageContainer image, final String encoderType, final float quality,
                            final boolean alpha ) throws ContentIOException, IOException {
+
+    if ( dataLocation == null ) {
+      return null;
+    }
+
```

Comment:
There aren't any unit test for this class and the current code structure doesn't allow for easy or "effective" testing, it would involve passing mocks to mocks and basically testing mocking framework and less so the actual pentaho logic.